src/main/groovy/org/shipkit/internal/gradle/git/GitSetupPlugin.java

```diff
@@ -54,22 +57,20 @@
     public void apply(Project project) {
         final ShipkitConfiguration conf = project.getPlugins().apply(ShipkitConfigurationPlugin.class).getConfiguration();
 
-        TaskMaker.execTask(project, UNSHALLOW_TASK, new Action<Exec>() {
-            public void execute(final Exec t) {
+        TaskMaker.task(project, UNSHALLOW_TASK, ShipkitExecTask.class, new Action<ShipkitExecTask>() {
```

Comment:
We could do that. I have not implemented it this way because the value is pretty small. Previously we had a factory method for Exec task because we needed to add an additional 'doFirst' action to all Exec tasks. With ShipkitExecTask, there is no need for this.